build_tools/cmake/presets/options.json

```diff
@@ -0,0 +1,149 @@
+                },
+                "IREE_TARGET_BACKEND_LLVM_CPU": {
+                    "type": "BOOL",
+                    "value": "OFF"
+                },
+                "IREE_TARGET_BACKEND_LLVM_CPU_WASM": {
+                    "type": "BOOL",
+                    "value": "OFF"
+                },
+                "IREE_TARGET_BACKEND_ROCM": {
+                    "type": "BOOL",
+                    "value": "OFF"
+                },
+                "IREE_TARGET_BACKEND_METAL_SPIRV": {
+                    "type": "BOOL",
+                    "value": "OFF"
+                },
+                "IREE_TARGET_BACKEND_VULKAN_SPIRV": {
+                    "type": "BOOL",
+                    "value": "OFF"
+                },
+                "IREE_TARGET_BACKEND_CUDA": {
+                    "type": "BOOL",
+                    "value": "OFF"
+                },
+                "IREE_TARGET_BACKEND_WEBGPU": {
+                    "type": "BOOL",
+                    "value": "OFF"
+                },
+                "IREE_INPUT_STABLEHLO": {
+                    "type": "BOOL",
+                    "value": "OFF"
+                },
+                "IREE_INPUT_TORCH": {
+                    "type": "BOOL",
+                    "value": "OFF"
+                },
+                "IREE_INPUT_TOSA": {
+                    "type": "BOOL",
+                    "value": "OFF"
+                },
+                "IREE_OUTPUT_FORMAT_C": {
+                    "type": "BOOL",
+                    "value": "OFF"
+                }
+            }
+        },
+        {
+            "name": "python-bindings",
+            "displayName": "Python Development",
+            "cacheVariables": {
+                "IREE_BUILD_PYTHON_BINDINGS": {
+                    "type": "BOOL",
+                    "value": "OFF"
+                },
+                "Python3_FIND_VIRTUALENV": {
+                    "type": "STRING",
+                    "value": "ONLY"
+                }
+            }
+        },
+        {
+            "name": "new-linux-minimal",
+            "displayName": "(START HERE) Set Up New Minimal Dev Config",
+            "description": "Project configured with minimal features enabled",
+            "inherits": [
+                "new-linux-dev",
+                "minimal"
+            ]
+        },
+        {
+            "name": "new-macos-minimal",
+            "displayName": "(START HERE) Set Up New Minimal Dev Config",
+            "description": "Project configured with minimal features enabled",
+            "inherits": [
+                "new-macos-dev",
+                "minimal"
+            ]
+        },
+        {
+            "name": "new-windows-minimal",
+            "displayName": "(START HERE) Set Up New Minimal Dev Config",
+            "description": "Project configured with minimal features enabled",
+            "inherits": [
+                "new-windows-dev",
+                "minimal"
+            ]
```

Comment:
Ah, nice job setting up the inheritance. I found that tricky here: https://github.com/openxla/iree/issues/5804#issuecomment-1461062980

Reply:
Yeah, I don't love the whole way inheritance interplays with conditions. The best I could come up with was to fork by platform on these "new" presets while having everything else be generic.